apps/meteor/server/models/raw/ModerationReports.ts

```diff
@@ -132,12 +135,97 @@ export class ModerationReportsRaw extends BaseRaw<IModerationReport> implements
 		return this.col.aggregate(params, { allowDiskUse: true });
 	}
 
+	findUserReports(
+		latest: Date,
+		oldest: Date,
+		selector: string,
+		pagination: PaginationParams<IModerationReport>,
+	): AggregationCursor<Pick<UserReport, '_id' | 'reportedUser' | 'ts'> & { count: number }> {
+		const query = {
+			_hidden: {
+				$ne: true,
+			},
+			ts: {
+				$lt: latest,
+				$gt: oldest,
+			},
+			message: {
+				$exists: false,
+			},
+			...this.getSearchQueryForSelectorUsers(selector),
+		};
+
+		const { sort, offset, count } = pagination;
+
+		const pipeline = [
+			{ $match: query },
+			{
+				$sort: {
+					ts: -1,
+				},
+			},
+			{
+				$group: {
+					_id: '$reportedUser._id',
+					count: { $sum: 1 },
+					reports: { $first: '$$ROOT' },
+				},
+			},
+			{
+				$sort: sort || {
+					'reports.ts': -1,
+				},
+			},
+			{
+				$skip: offset,
+			},
+			{
+				$limit: count,
+			},
+			{
+				$project: {
+					_id: 0,
+					reportedUser: '$reports.reportedUser',
+					ts: '$reports.ts',
+					count: 1,
+				},
+			},
+		];
+
+		return this.col.aggregate(pipeline, { allowDiskUse: true, readPreference: readSecondaryPreferred() });
+	}
+
 	countMessageReportsInRange(latest: Date, oldest: Date, selector: string): Promise<number> {
-		return this.col.countDocuments({
-			_hidden: { $ne: true },
-			ts: { $lt: latest, $gt: oldest },
-			...this.getSearchQueryForSelector(selector),
-		});
+		return this.col
+			.distinct('message.u._id', {
+				_hidden: {
+					$ne: true,
+				},
+				ts: {
+					$lt: latest,
+					$gt: oldest,
+				},
+				...this.getSearchQueryForSelector(selector),
+			})
+			.then((ids) => ids.length);
+	}
+
+	countUserReportsInRange(latest: Date, oldest: Date, selector: string): Promise<number> {
+		return this.col
+			.distinct('reportedUser._id', {
```

Comment:
this also i don't understand. counting user reports means all user reports .. not unique ones by reporter. if I report two users that will count to one?

Reply:
Ok, same thing as before. Please update the method name. Because now it isn't really counting user reports - user reports mean ALL individual reports for any user. Rather something like `totalUsersReportedCountInRange` (mind the `ed`).

Reply:
Thank you, done!